django_project/geosight/data/serializer/related_table.py

```diff
@@ -23,8 +25,33 @@
 )
 
 
+class RelatedTableApiSerializer(DynamicModelSerializer):
+    url = serializers.SerializerMethodField()
+    creator = serializers.SerializerMethodField()
+    fields_definition = serializers.SerializerMethodField()
+
+    def get_url(self, obj: RelatedTable):
+        return reverse(
+            'related-tables-detail',
+            args=[obj.id]
+        )
+
+    def get_creator(self, obj: RelatedTable):
+        return obj.creator.get_full_name() if obj.creator else None
+
+    def get_fields_definition(self, obj: RelatedTable):
```

Comment:
Right now it's returning the same `fields_definition` as in the other endpoint that feeds the ui (read-only, without `label`, etc.). This is the main reason I'm marking the PR as draft, since we need to have a means of persisting/obtaining that information on the database for this to work as defined in the issue.

Reply:
Yes, we can keep like this for now
I will update the model and code accordingly